tests/tensor/test_max_argmax.py

```diff
@@ -0,0 +1,693 @@
+                    data,
+                    eval_outputs(grad(max_and_argmax(n, axis=axis)[0].sum(), n)),
+                    axis=axis,
+                )
+            check_grad_max(data, eval_outputs(grad(max_and_argmax(n.flatten())[0], n)))
+
+        # Test 3d inner dimensions
+        data = random(3, 4, 5)
+
+        for i in [0, 1, 2]:
+            safe_verify_grad(lambda v: max_and_argmax(v, axis=[i])[0], [data])
+            safe_verify_grad(lambda v: max_and_argmax(v, axis=[i])[1], [data])
+
+        # Test 4d inner dimensions
+        data = random(2, 3, 4, 5)
+
+        for i in [0, 1, 2, 3]:
+            safe_verify_grad(lambda v: max_and_argmax(v, axis=[i])[0], [data])
+            safe_verify_grad(lambda v: max_and_argmax(v, axis=[i])[1], [data])
+
+        # Test grad with multiple axes
+        for i in [[0, 1], [0, 0]]:
+            safe_verify_grad(lambda v: max_and_argmax(v, axis=i)[0], [data])
+            safe_verify_grad(lambda v: max_and_argmax(v, axis=i)[1], [data])
+
+    def test_preserve_broadcastable(self):
+        # Ensure the original broadcastable flags are preserved by Max/Argmax.
+        x = matrix().dimshuffle("x", 0, "x", 1, "x")
+        y = x.max(axis=1)
+        assert y.type.shape == (1, 1, None, 1)
+        assert y.type.broadcastable == (True, True, False, True)
+
+    def test_multiple_axes(self):
+        data = np.arange(24).reshape(3, 2, 4)
+        x = as_tensor_variable(data)
+        vt, it = max_and_argmax(x, [1, -1])
+        assert vt.type.shape == it.type.shape == (3,)
+        v, i = eval_outputs([vt, it])
+        assert np.all(v == np.array([7, 15, 23]))
+        assert np.all(i == np.array([7, 7, 7]))
+        v = eval_outputs(vt.shape)
+        assert tuple(v) == vt.type.shape
+
+    def test_zero_shape(self):
+        x = matrix()
+        m, i = max_and_argmax(x, axis=1)
+        f = function([x], [m, i])
+        xv = np.zeros((0, 4), dtype=config.floatX)
+        mv, iv = f(xv)
+        assert mv.shape == (0,)
+        assert iv.shape == (0,)
+
+    def test_numpy_input(self):
+        ar = np.array([1, 2, 3])
+        max_pt, argmax_pt = max_and_argmax(ar, axis=None)
+        assert max_pt.eval() == 3
+        assert argmax_pt.eval() == 2
+
+    @pytest.mark.parametrize(
+        "core_axis, batch_axis",
+        [
+            (None, (1, 2, 3, 4)),
+            (0, (1,)),
+            ((1, -1), (2, 4)),
+        ],
+    )
+    def test_vectorize(self, core_axis, batch_axis):
+        x = tensor(shape=(5, 5, 5, 5))
+        batch_x = tensor(shape=(3, 5, 5, 5, 5))
+
+        # Test MaxAndArgmax
+        max_x, argmax_x = max_and_argmax(x, axis=core_axis)
+        node = max_x.owner
+        assert isinstance(node.op, TensorMax)
+
+        # dbt: how to make Argmax user facing?
+        # new_node = vectorize_node(node, batch_x)
+        # pytensor.dprint(new_node)
+        # assert isinstance(new_node.op, Argmax)
+        # assert new_node.op.axis == batch_axis
+
```

Comment:
How to make Argmax user facing?

Reply:
Argmax was already user facing under `pt.argmax()`

Reply:
Ohk.
I was actually trying to have a test along this lines for ```Argmax```:
https://github.com/pymc-devs/pytensor/blob/d34760d74b9b7544aa50efb63c3be27a8bd400ff/tests/tensor/test_math.py#L1038-L1040

However, the current implementation for ```Argmax``` is tested using:
https://github.com/pymc-devs/pytensor/blob/d34760d74b9b7544aa50efb63c3be27a8bd400ff/tests/tensor/test_math.py#L1042-L1048

Here 1043 states that Argmax is not user facing so I thought this might be causing the previous approach to fail.

Reply:
Ah, it didn't use to be user-facing (the MaxAndArgmax was the thing users saw), but will be after this PR